Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/DownloadDiagnosisKeysTask.kt

```diff
@@ -174,6 +174,12 @@ class DownloadDiagnosisKeysTask @Inject constructor(
 
         val hasRecentDetection = nextForcedDetectionAt != null && now.isBefore(nextForcedDetectionAt)
 
+        val lastSuccesfulDetectionInFuture = nextForcedDetectionAt?.isBefore(now)
```

Comment:
The check seems wrong, shouldn't it be:

```kotlin
val lastSuccesfulDetectionInFuture = lastSuccessfulDetection?.startedAt?.isAfter(now) ?: false
```

* If `lastSuccessfulDetection` is in the distant past (time travel backwards), then `hasRecentDetection` is false.
* If `lastSuccessfulDetection` is in the distant future (time travel forwards), then `now.isBefore(nextForcedDetectionAt)` is always true, and for more than the 1 day we add, causing this to be `true` until the present has caught up with the future.
* But `lastSuccessfulDetection?.startedAt` should never be after or equal to the current time, except if time travel into the future occurs.